### PR TITLE
[pipeline] set optimizer to optional in execute_pipeline

### PR DESCRIPTION
**Pull request by Fridge003**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

- set optimizer argument to optional in execute_pipeline, since optimizer is not needed in evaluation
- move logging message to dist_logger when initializing booster

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/booster/booster.py [hunk 1]
```diff
@@ -64,17 +65,20 @@ def __init__(self,
             assert isinstance(
                 plugin, Plugin), f'Expected the argument plugin to be an instance of Plugin, but got {type(plugin)}.'
         self.plugin = plugin
+        logger = get_dist_logger()
```
Comment (ver217) [inline colossalai/booster/booster.py:68]: DON'T use this, as it relies on gpc and gpc will be deprecated soon.

File: colossalai/booster/booster.py [hunk 2]
```diff
@@ -64,17 +65,20 @@ def __init__(self,
...
-            warnings.warn('The plugin will control the accelerator, so the device argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
+                           ranks=[0])
         else:
             self.accelerator = Accelerator(device)
 
         # set precision
         if self.plugin and self.plugin.control_precision():
-            warnings.warn('The plugin will control the precision, so the mixed_precision argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
```
Comment (ver217) [inline colossalai/booster/booster.py:80]: Only warn when `mixed_precision` is not None

File: colossalai/booster/booster.py [hunk 3]
```diff
@@ -64,17 +65,20 @@ def __init__(self,
             assert isinstance(
                 plugin, Plugin), f'Expected the argument plugin to be an instance of Plugin, but got {type(plugin)}.'
         self.plugin = plugin
+        logger = get_dist_logger()
 
         # set accelerator
         if self.plugin and self.plugin.control_device():
             self.accelerator = None
-            warnings.warn('The plugin will control the accelerator, so the device argument will be ignored.')
+            logger.warning('The plugin will control the accelerator, so the device argument will be ignored.',
```
Comment (ver217) [inline colossalai/booster/booster.py:73]: Can we set default value of `device`  arg of this function to `None`? Then we can only warn if `device` is not None here. And if no plugin is used or plugin does not control device, we can set `device = device or 'cuda'`. In the docstring, we have to declare the default value is cuda. @FrankLeeeee

File: colossalai/booster/booster.py (hunk 1 above)
Comment (Fridge003) [reply colossalai/booster/booster.py]: removed

File: colossalai/booster/booster.py (hunk 2 above)
Comment (Fridge003) [reply colossalai/booster/booster.py]: modified

File: colossalai/booster/booster.py (hunk 3 above)
Comment (Fridge003) [reply colossalai/booster/booster.py]: modified

**Review by ver217 — APPROVED** [review]

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**